internal/controller/rbac/definition/roles.go

```diff
@@ -99,6 +101,15 @@ func RenderClusterRoles(d *v1.CompositeResourceDefinition) []rbacv1.ClusterRole
 				Resources: []string{d.Spec.Names.Plural},
 				Verbs:     verbsEdit,
 			},
+			// The rule grants access to update the finalizers of resources that belong to the API group.
+			// If OwnerReferencesPermissionEnforcement is enabled, it allows after deleting the XRD to
+			// clean up resources that they no longer own, so that the system can be kept in a consistent state.
+			// See https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
+			{
+				APIGroups: []string{d.Spec.Group},
+				Resources: []string{d.Spec.Names.Plural + suffixFinalizers},
+				Verbs:     verbsUpdate,
+			},
```

Comment:
Are you sure we want the edit role to have these permissions and not the system role?

* https://github.com/crossplane/crossplane/blob/0951db/cluster/charts/crossplane/templates/clusterrole.yaml#L12

I'm a bit rusty on the RBAC manager, but I'm pretty sure the Crossplane pod ran as the system role (not edit, which is for human users) - see `clusterrole.yaml` above. I'm guessing the intent here is that Crossplane should be allowed to create composed resources that block the deletion of their XRD.